lms/djangoapps/progress/signals.py

```diff
@@ -16,9 +19,25 @@
 from edx_notifications.data import NotificationMessage
 
 from progress.models import StudentProgress, StudentProgressHistory, CourseModuleCompletion
+
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
 log = logging.getLogger(__name__)
 
 
+def _get_parent_content_id(html_content_id):
+    try:
+        html_usage_id = BlockUsageLocator.from_string(html_content_id)
+        html_module = modulestore().get_item(html_usage_id)
+        parent_module = html_module.get_parent()
+        return str(parent_module.scope_ids.usage_id)
```

Comment:
It would be much more efficient to just `return html_module.parent`, which unless I'm mistaken is the same as `html_module.get_parent().scope_ids.usage_id` but doesn't need to load the parent (which is very expensive).